sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -290,12 +305,13 @@ private void resetPlanner()
   private PlannerResult planWithDruidConvention(
       final RelRoot root,
       @Nullable final SqlExplain explain,
-      @Nullable final SqlInsert insert
+      @Nullable final SqlInsert insert,
```

Comment:
I echo with Rohan's comment here. If we can help it we should try and collapse insert and replace into a single variable. The only time distinction would be necessary would be when we are trying to populate the context parameters with the time chunks/clustered by/partitioned by etc. Otherwise, the underlying engine doesn't need to know about the presence of two separate nodes.

Reply:
Done. Type is only checked during creation of ParsedNode and to add it to the context.